activate.go

```diff
@@ -675,6 +698,38 @@ func (m *activateOneContainerStateMachine) primaryKeyInfo() (PrimaryKey, keyring
 	return m.primaryKey, m.primaryKeyID, nil
 }
 
+func (m *activateOneContainerStateMachine) activationState() (*ContainerActivateState, error) {
+	if m.hasMoreWork() {
+		return nil, errors.New("state machine has not finished")
```

Comment:
this little bit is not exercised

Reply:
There's a similar path in another function that isn't tested as well. I've added a couple of tests for both of them now.